rcli/call.py

```diff
@@ -0,0 +1,406 @@
+# -*- coding: utf-8 -*-
+"""A module for calling subcommands and handling type hint casting.
+
+Functions:
+    call: Calls a function and casts the docopt-style args to the appropriate
+        names and types.
+    get_callable: Retrieves a callable object from a subcommand.
+"""
+
+from __future__ import unicode_literals
+
+import collections
+import csv
+import functools
+import inspect
+import json
+import logging
+import re
+import sys
+import tokenize
+import types
+import typing
+
+import six
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+_CSV_TYPES = (
+    list,
+    tuple,
+    set,
+    frozenset,
+    collections.deque,
+    collections.Counter,
+    typing.MutableSequence,
+    typing.MutableSet,
+    typing.Tuple
```

Comment:
This has to be added in addition to `tuple` because `typing.Tuple` does not extend `tuple` in Python 3.5.2.